File: Arrays/LeftRotation/LeftRotation.py

import math


class LeftRotation:
	
	n = d = 0
	n_arr = []
	def __init__(self,n_arr_in, n_in, d_in):
		self.n = n_in
		self.d = d_in
		self.n_arr = n_arr_in

	def solve(self):

		print self.n
		print self.d
		print self.n_arr
		temp = self.n_arr[self.d:len(self.n_arr)]
		temp.extend(self.n_arr[0:self.d])

		return temp



def main():
	f = open("input.txt");
	nd = []
	n = []
	
	nd = ([int(i) for i in f.readline().rstrip().split()])
	n_arr = ([int(i) for i in f.readline().rstrip().split()])
	#print(nd);
	#print(n);
	print(LeftRotation(n_arr,nd[0],nd[1]).solve())
main()

File: Arrays/LeftRotation/LeftRotation.cpp

#include <iostream>
#include <fstream>
#include <string.h>
#include <sstream>
#include <vector> 


using namespace std;


void printvector(vector<int> arr_in) {
	for(size_t i = 0; i < arr_in.size();i++) {		
		cout << arr_in[i];
		cout << endl;
	}

}
class LeftRotation
{
	private:
		int n, d;
		vector<int> arr;
	public:
		LeftRotation(int d_in, vector<int> arr_in)
		{
			d = d_in;
			arr = arr_in;
		}
		vector<int> solve()
		{
			vector<int> answer(arr.size());
			int d_mod = d % arr.size();
			int x = 0;

			for (int i = d_mod;i<arr.size(); i++)
			{
				answer[x] = arr[i];
				x++;
			}

			for (int i = 0;i<d_mod;i++)
			{
				answer[x] = arr[i];
				x++;
			}
			return answer;

		}

};

int main()
{
	fstream myfile;
	myfile.open("input.txt");
	string line, row;
	vector<int> arr;
	int n, d = 0;
	if (myfile.is_open()) {
		
		getline(myfile, row, '\n');
		stringstream ssin(row);
		string word;
		ssin >> word;
		n = atoi(word.c_str());
		ssin >> word;
		d = atoi(word.c_str());

		getline(myfile, row, '\n');
		ssin.str(row);
		ssin.clear();
		arr.resize(n);
		int i = 0;

		while(ssin.good()) {
			ssin >> word;
			arr[i] = atoi(word.c_str());
			i++;
		}
	}
	//printvector(arr);
	printvector((new LeftRotation(d, arr))->solve());
	myfile.close();
}